src/main/java/io/supertokens/passwordless/Passwordless.java

```diff
@@ -124,6 +129,121 @@ private static String generateUserInputCode() {
+                            passwordlessStorage.commitTransaction(con);
+                            throw new StorageTransactionLogicException(new RestartFlowException());
+                        } else {
+                            passwordlessStorage.incrementDeviceFailedAttemptCount_Transaction(con, deviceIdHash);
+                            passwordlessStorage.commitTransaction(con);
+
+                            if (code != null) {
+                                throw new StorageTransactionLogicException(new ExpiredUserInputCodeException(
+                                        device.failedAttempts + 1, maxCodeInputAttempts));
+                            } else {
+                                throw new StorageTransactionLogicException(new IncorrectUserInputCodeException(
+                                        device.failedAttempts + 1, maxCodeInputAttempts));
+                            }
+                        }
+                    }
+                    throw new StorageTransactionLogicException(new RestartFlowException());
+                }
+
+                if (device.email != null) {
+                    passwordlessStorage.deleteDevicesByEmail_Transaction(con, device.email);
+                } else if (device.phoneNumber != null) {
+                    passwordlessStorage.deleteDevicesByPhoneNumber_Transaction(con, device.phoneNumber);
+                }
+
+                passwordlessStorage.commitTransaction(con);
+                return device;
+            });
+        } catch (StorageTransactionLogicException e) {
+            if (e.actualException instanceof ExpiredUserInputCodeException) {
+                throw (ExpiredUserInputCodeException) e.actualException;
+            }
+            if (e.actualException instanceof IncorrectUserInputCodeException) {
+                throw (IncorrectUserInputCodeException) e.actualException;
+            }
+            if (e.actualException instanceof RestartFlowException) {
+                throw (RestartFlowException) e.actualException;
+            }
+            throw e;
+        }
+
+        UserInfo user = consumedDevice.email != null ? passwordlessStorage.getUserByEmail(consumedDevice.email)
+                : passwordlessStorage.getUserByPhoneNumber(consumedDevice.phoneNumber);
+        if (user == null) {
+            while (true) {
+                try {
+                    if (user == null) {
+                        String userId = Utils.getUUID();
+                        long timeJoined = System.currentTimeMillis();
+                        user = new UserInfo(userId, consumedDevice.email, consumedDevice.phoneNumber, timeJoined);
+                        passwordlessStorage.createUser(user);
+                        return new ConsumeCodeResponse(true, user);
+                    }
+                } catch (DuplicateEmailException | DuplicatePhoneNumberException e) {
+                    throw new RestartFlowException();
```

Comment:
Write why we are throwing `RestartFlowException` instead of signing in the user. I mean an explanation of a scenario in which it would come here and what the UX will be at that time.